### PR TITLE
Allowing YARD to parse content of C++ namespaces. Fixes #809.

### LINKED ISSUE #809: C-Parser: cant find method definition when using namespaces
SampleCode

```c++
VALUE rb_cWXRect;

namespace RubyWX {
namespace Rect {

VALUE _alloc(VALUE self)
{
	return wrap(new wxRect());
}

/*
 * call-seq:
 *   inspect -> String
 *
 * Human-readable description.
 * ===Return value
 * String
*/
VALUE _inspect(VALUE self)
{
	return rb_sprintf( "%s(%d, %d, %d, %d)",
		rb_obj_classname( self ),
		FIX2INT(_getX(self)),
		FIX2INT(_getY(self)),
		FIX2INT(_getWidth(self)),
		FIX2INT(_getHeight(self)));
}

}
}

void Init_WXRect()
{
	using namespace RubyWX::Rect;
	rb_cWXRect = rb_define_class("Rect",rb_cObject);

	rb_define_alloc_func(rb_cWXRect,_alloc);

	rb_define_method(rb_cWXRect,"inspect",RUBY_METHOD_FUNC(_inspect),0);

}
```

returns this
```
Files:           2
Modules:         0 (    0 undocumented)
Classes:         1 (    1 undocumented)
Constants:       0 (    0 undocumented)
Methods:         1 (    1 undocumented)
 0.00% documented
```
currenty the Files2 is a bug that happens because one of the o.file is nil
but for my self i cant fix it that it does find the method with the c++ namespaces

### PR DESCRIPTION
**Pull request by thomthom**
# Description

This Pull Request attempts to address #809 - where content inside of C++ namespaces isn't processed. This is an issue for source code that use the Ruby C API in a C++ environment.

What this fix do is simply make the C Parser not treat the content of namespaces as body statements. Instead it just ignores the namespaces all together.

I'm not sure if it's perfect, but we're working on getting a very sizable amount of code parsed so I'll follow up with improvements as needed if we run into them.

# Completed Tasks

* [x] I have read the [Contributing Guide][contrib].
* [x] The pull request is complete (implemented / written).
* [x] Git commits have been cleaned up (squash WIP / revert commits).
* [x] I wrote tests and ran `bundle exec rake` locally (if code is attached to PR).

[contrib]: https://github.com/lsegal/yard/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by thomthom** [comment]
I had to add another commit to correct a last moment issue that made RuboCop fail. Do I have to go back and squash this? Or can you just use the GitHub Merge and Squash feature?

I was at first not running rubocop before I submitted because I threw errors on all files due to line endings. That's because git normalize to OS line endings upon checkout - but then correct them when you check in.

I ended up adding these lines temporarily to `.rubocop` to mute those warnings:

```
EndOfLine:
  Enabled: false
```

Since git normalize line ending upon checking, is that RuboCop check really needed?

**Comment by thomthom** [comment]
hmm... hold this PR. I might have run into a regression. Let me run more tests to be sure.

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 source 'https://rubygems.org'
 
 group :development do
-  gem 'rspec', '~> 3.0'
+  gem 'rspec', '~> 3.5'
```
Comment (thomthom) [inline Gemfile]: Ah, so I had not intended this to be part of this PR. But this was something I had to do in order to get things running. See #1062

File: lib/yard/parser/c/c_parser.rb [hunk 2]
```diff
@@ -38,6 +38,7 @@ def parse_toplevel
             when '#'; consume_directive
             when '/'; consume_comment
             when /\s/; consume_whitespace
+            when '}'; advance # Skip possible C++ namespace closing brackets.
```
Comment (thomthom) [inline lib/yard/parser/c/c_parser.rb]: So this one, I'm not sure if you'll like this much. I can try to refactor it if you prefer. Though any suggestions would be appreciated. The issue here was that when there are namespaces my fix was to just ignore it. But then `consume_toplevel_statement` would later run into stray `}` (closing namespaces) that would throw it off.

So this line is currently very generic in advancing over any `}` in the top level. I could try to keep track of the depth of the C++ namespaces and only advance over `}` a matching number of times.

It all depend on how low impact we want to keep this. I ran this fix over our code base with over 1300 methods and 98 classes - no documentation objects where lost.

File: Gemfile (hunk 1 above)
Comment (lsegal) [inline Gemfile]: Undo this change and rebase off of master since this fix was pushed separately.

Comment (thomthom) [inline Gemfile]: Ok. So when rebasing, do I then create a new PR? since that changes the history of the branch. (I'm not that familiar with rebase.)

File: lib/yard/parser/c/c_parser.rb (hunk 2 above)
Comment (lsegal) [inline lib/yard/parser/c/c_parser.rb]: I'm not too concerned, since the worst case is that it continues past a toplevel element to find the next one.
